api/src/test/scala/hmda/api/processing/HmdaFileUploadSpec.scala

```diff
@@ -24,6 +25,27 @@ class HmdaFileUploadSpec extends ActorSpec {
       probe.send(hmdaFileUpload, GetState)
       probe.expectMsg(HmdaFileUploadState(Map(timestamp -> 4)))
     }
-  }
 
+    "recover with event" in {
+      probe.send(hmdaFileUpload, Shutdown)
+
+      val secondHmdaFileUpload = createHmdaFileUpload(system, "1")
+
+      probe.send(secondHmdaFileUpload, GetState)
+      probe.expectMsg(HmdaFileUploadState(Map(timestamp -> 4)))
+    }
+
+    "recover with from snapshot" in {
+      val thirdHmdaFileUpload = createHmdaFileUpload(system, "1")
+      probe.send(thirdHmdaFileUpload, CompleteUpload)
+      probe.send(thirdHmdaFileUpload, Shutdown)
+
+      Thread.sleep(500) //wait for actor messages to be processed so that the state can be saved
```

Comment:
Make sure to shutdown `thirdHmdaFileUpload` before starting `fourthHmdaFileUpload`.